### ISSUE TITLE
CSmartTable itemsPerPage selection changes CSmartPagination limit prop too

### ISSUE THREAD

**Issue by dang-cb** [coreui/coreui-vue#180]
The CSmartTable component uses the CSmartPagination component to display page navigation, but it wires up its `limit` prop to be the user's `itemsPerPage.value` selection, which means it gets updated every time the user selects a different items per page count.

This is really noticeable with a large table, say with 1,500 items.
With Items per page set to "5", the CSmartPagination shows `<< 1, 2, 3, 4 ... >>`
Now select "10", and it will change to `<< 1, 2, 3, 4, 5, 6, 7, 8, 9 ... >>`
Switch to 50, and it changes to `<< 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 27, 28, 29, 30 >>`
(Needless to say, the rendering width will break the container and probably introduce horizontal scroll bars in the browser)

### The fix:
In _CSmartTable.ts_ where CSmartPagination is wired up, simply remove the setting of `limit`, i.e.
```
? h(CSmartPagination, {
    ...props.paginationProps,
    pages: numberOfPages.value,
    activePage: activePage.value,
    onActivePageChange: handleActivePageChange,
    // limit: itemsPerPage.value,
  })
```
limit will default to 5, but can now also be set (and no longer overridden) directly from our CSmartTable definition, e.g. to set the pagination limit to 10 we can:
```
<CSmartTable pagination items-per-page-select :items="items" :columns="columns" :pagination-props="{limit:10}">
```

**Comment by mrholek** [comment]
Thank you for your help, I will fix it next release.

**Comment by dang-cb** [comment]
(Tested but not fixed yet in v4.1.2)

**Comment by mrholek** [comment]
Please check if you still have this in v4.1.3

**Comment by dang-cb** [comment]
Confirmed fixed in v4.1.3, thanks once again @mrholek 💯
